### PR TITLE
My account to backend

### PR DESCRIPTION
**Pull request by elacymerys**

### CODE REVIEW

File: thesis-gui/src/components/my-account/UserRanking.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            </IonLabel>
+        </IonItem>
+    );
+}
+
+export const UserRanking = () => {
+    const { user } = useUserContext();
+    const { categories } = useCategoryContext();
+
+    const getCategoryNameWithId = (id: number) => {
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-account/UserRanking.tsx:30]: I think this whole function can be simplified to:
`return categories.find(category => category.id === id)?.name || '';`

File: thesis-gui/src/context/CategoryContext.tsx [hunk 2]
```diff
@@ -36,8 +36,10 @@ export const CategoryContextProvider: FC = ({ children }) => {
     useEffect(() => {
         if (!user) {
```
Comment (MaksymilianK) [inline thesis-gui/src/context/CategoryContext.tsx]: I would swap this `if else`:
```
if (user) {
  getCategories();
} else {
  setChosenCategories([]);
}
```

I think it's more readable when there is no negation in `if`.

**Review by MaksymilianK — CHANGES_REQUESTED** [review]

File: thesis-gui/src/components/my-account/UserRanking.tsx (hunk 1 above)
Comment (elacymerys) [reply thesis-gui/src/components/my-account/UserRanking.tsx:30]: Done

File: thesis-gui/src/context/CategoryContext.tsx (hunk 2 above)
Comment (elacymerys) [reply thesis-gui/src/context/CategoryContext.tsx]: Done

**Review by MaksymilianK — APPROVED** [review]

**Pull request merged**